sound/soc/intel/boards/sof_cs42l42.c

```diff
@@ -37,7 +37,19 @@
 #define SOF_CS42L42_NUM_HDMIDEV_MASK		(GENMASK(9, 7))
 #define SOF_CS42L42_NUM_HDMIDEV(quirk)	\
 	(((quirk) << SOF_CS42L42_NUM_HDMIDEV_SHIFT) & SOF_CS42L42_NUM_HDMIDEV_MASK)
-#define SOF_MAX98357A_SPEAKER_AMP_PRESENT	BIT(10)
+#define SOF_CS42L42_DAILINK_SHIFT		10
+#define SOF_CS42L42_DAILINK_MASK		(GENMASK(24, 10))
+#define SOF_CS42L42_DAILINK(link1, link2, link3, link4, link5) \
+	((((link1) | (link2 << 3) | (link3 << 6) | (link4 << 9) | (link5 << 12)) << SOF_CS42L42_DAILINK_SHIFT) & SOF_CS42L42_DAILINK_MASK)
+#define SOF_MAX98357A_SPEAKER_AMP_PRESENT	BIT(25)
+
+enum {
+	LINK_NONE = 0,
```

Comment:
I am not clear on what this means.

Is this an error case, as I read below
````c
case LINK_NONE:
		default:
			dev_err(dev, "invalid link type %d\n", link_type);
			goto devm_err;
````
Is this used as a 'terminator', e.g. so that you could SPK, DMIC, HDMI and no HP?
Is this a placeholder for future extensions?

Reply:
Like you said, LINK_NONE is used as terminator in the SOF_CS42L42_DAILINK macro. The 'case LINK_NONE' here is for the case that LINK_NONE is not used as terminator like this:

SOF_CS42L42_DAILINK(LINK_HP, LINK_DMIC, LINK_HDMI, LINK_NONE, LINK_SPK),

Reply:
Thanks @brentlu, that's quite clever indeed - and that shows I am not :-)

maybe adding some comments would help, just to clarify the intent and error cases.

Reply:
Will do. It also took me a while to figure out why there is a "case LINK_NONE:" there...